contributors/design-proposals/encryption.md

```diff
@@ -0,0 +1,436 @@
+# Encryption
+
+## Abstract
+
+The scope of this proposal is to ensure that resources can be encrypted at the
+datastore layer with sufficient metadata support to enable integration with
+multiple encryption providers and key rotation. Encryption will be optional for
+any resource, but will be used by default for the Secret resource. Secrets are
+already protected in transit via TLS.
+
+Full disk encryption of the volumes storing etcd data is already expected as
+standard security hygiene.  Adding the proposed encryption at the datastore
+layer defends against malicious parties gaining access to:
+
+- The etcd API
```

Comment:
I believe the point here is encrypting the data at rest within etcd defends against someone who has direct etcd access from listing the secrets. It is not actually about protecting the etcd API.